### PR TITLE
add quaternion to euler conversion

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

implements euler angles to quaternion and viceversa

https://gist.github.com/salmagro/2e698ad4fbf9dae40244769c5ab74434

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by cjpurackal** [comment]
This seems to work up till pi/2, after which it fails. Not sure if this is expected behavior 

```
roll = pitch = yaw = torch.linspace(0, torch.pi, 10)
roll
tensor([0.0000, 0.3491, 0.6981, 1.0472, 1.3963, 1.7453, 2.0944, 2.4435, 2.7925,
        3.1416])
qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
roll_new, pitch_new, yaw_new = euler_from_quaternion(qw, qx, qy, qz)
roll_new
tensor([ 0.0000e+00,  3.4907e-01,  6.9813e-01,  1.0472e+00,  1.3963e+00,
        -1.3963e+00, -1.0472e+00, -6.9813e-01, -3.4907e-01,  8.7423e-08])

```

**Comment by edgarriba** [comment]
> This seems to work up till pi/2, after which it fails. Not sure if this is expected behavior
> 
> ```
> roll = pitch = yaw = torch.linspace(0, torch.pi, 10)
> roll
> tensor([0.0000, 0.3491, 0.6981, 1.0472, 1.3963, 1.7453, 2.0944, 2.4435, 2.7925,
>         3.1416])
> qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
> roll_new, pitch_new, yaw_new = euler_from_quaternion(qw, qx, qy, qz)
> roll_new
> tensor([ 0.0000e+00,  3.4907e-01,  6.9813e-01,  1.0472e+00,  1.3963e+00,
>         -1.3963e+00, -1.0472e+00, -6.9813e-01, -3.4907e-01,  8.7423e-08])
> ```

Yeah, I noticed that. Need to verify what's going on, I'll add yours as test too

**Comment by Hackerman342** [comment]
> This seems to work up till pi/2, after which it fails. Not sure if this is expected behavior
> 
> ```
> roll = pitch = yaw = torch.linspace(0, torch.pi, 10)
> roll
> tensor([0.0000, 0.3491, 0.6981, 1.0472, 1.3963, 1.7453, 2.0944, 2.4435, 2.7925,
>         3.1416])
> qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
> roll_new, pitch_new, yaw_new = euler_from_quaternion(qw, qx, qy, qz)
> roll_new
> tensor([ 0.0000e+00,  3.4907e-01,  6.9813e-01,  1.0472e+00,  1.3963e+00,
>         -1.3963e+00, -1.0472e+00, -6.9813e-01, -3.4907e-01,  8.7423e-08])
> ```
@cjpurackal 
This is the values wrapping around 2*pi. So you start with a vector on the range [0,pi] and end with one on the range [-pi,pi].

**Comment by Hackerman342** [comment]
So recommend you test with:
```
roll = pitch = yaw = torch.linspace(-torch.pi, torch.pi, 20)
```

**Comment by edgarriba** [comment]
@cjpurackal added test againts `transforms3d` and all seems alright.

@stevenlovegrove you might want to give a look in the implementation

**Review by ducha-aiki — APPROVED** [review]

File: test/geometry/test_conversions.py [hunk 1]
```diff
@@ -1631,3 +1634,131 @@ def test_everything(self, device, dtype):
...
+
+    def test_forth_and_back(self, device, dtype):
+        roll, pitch, yaw = torch.rand(3, 2, device=device, dtype=dtype)
+        qw, qx, qy, qz = quaternion_from_euler(roll, pitch, yaw)
+        roll_new, pitch_new, yaw_new = euler_from_quaternion(qw, qx, qy, qz)
+        assert_close(roll, roll_new)
+        assert_close(pitch, pitch_new)
+        assert_close(yaw, yaw_new)
+
+    def test_values(self, device, dtype):
```
Comment (cjpurackal) [inline test/geometry/test_conversions.py:1732]: I'm getting some failed runs here when testing on cuda.

Comment (edgarriba) [reply test/geometry/test_conversions.py:1732]: now is fixed

**Pull request merged**